packages/react-components/src/components/table-output-component.tsx

```diff
@@ -524,14 +578,19 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
             });
 
             if (isFound) {
+                // match found in current block
                 return;
             }
 
-            const currRowIndex = this.gridApi?.getLastDisplayedRow() + 1;
-            const lineIndex = await this.findNextMatchIndex(currRowIndex);
-            if (lineIndex) {
+            // find match outside the current block
+            const lineIndex = await this.findMatchIndex(currRowIndex, direction);
```

Comment:
This sends the index of a row that should not be searched, then the server skips that index and starts at the next/previous index.
I propose that the client should send the first index that should be searched (so after/before the selection), and the server be changed to start at that index. See https://git.eclipse.org/r/c/tracecompass/org.eclipse.tracecompass/+/183874.